src/shared/watchdog.c

```diff
@@ -54,6 +55,7 @@ static int update_timeout(void) {
                                 return -errno;
                 }
 
+                watchdog_last_ping = now(clock_boottime_or_monotonic());
```

Comment:
Should this be set after watchdog ping is successful on the off chance there's a failure to ping the hardware watchdog?

Reply:
I used the patch to reduce the ioctl calls as much as possible and since the return value for watchdog_ping() is ignored everywhere there is no explicit action for retry. But I agree, it make sense to update watchdog_last_ping only if the heartbeat was successful to increase the chances of a successful heartbeat in case of some strange watchdog driver state. I will update the patch.